src/sinol_make/commands/run/__init__.py

```diff
@@ -0,0 +1,542 @@
+# Modified version of https://sinol3.dasie.mimuw.edu.pl/oij/jury/package/-/blob/master/runner.py
+# Author of the original code: Bartosz Kostka <kostka@oij.edu.pl>
+# Version 0.6 (2021-08-29)
+
+from sinol_make.interfaces.BaseCommand import BaseCommand
+from sinol_make.interfaces.Errors import CompilationError
+from sinol_make.helpers import compile, compiler
+import sinol_make.util as util
+import yaml, os, collections, sys, re, math
+import multiprocessing as mp
+
+class Command(BaseCommand):
+	"""
+	Class for running current task
+	"""
+
+
+	def get_name(self):
+		return 'run'
+
+
+	def configure_subparser(self, subparser):
+		parser = subparser.add_parser(
+			'run', 
+			help='Run current task',
+			description='Run current task'
+		)
+		parser.add_argument('--programs', type=str, nargs='+',
+							help='programs to be run, for example prog/abc{b,s}*.{cpp,py}')
+		parser.add_argument('--tests', type=str, nargs='+',
+							help='tests to be run, for example in/abc{0,1}*')
+		parser.add_argument('--cpus', type=int,
+							help='number of cpus to use, you have %d avaliable' % mp.cpu_count())
+		parser.add_argument('--tl', type=float, help='time limit (in s)')
+		parser.add_argument('--ml', type=float, help='memory limit (in MB)')
+		parser.add_argument('--verbose', dest='verbose', action='store_true',
+							help='show detailed raport (status for every test in a group)')
+		parser.add_argument('--show_memory', dest='show_memory', action='store_true',
+							help='show memory usage in report')
+		parser.add_argument('--validate_subtasks',
+							dest='validate_subtasks', action='store_true',
+							help='validate subtasks defined in config.yml')
+		parser.add_argument('--validate_programs',
+							dest='validate_programs', action='store_true',
+							help='validate programs in prog/ directory')
+		parser.add_argument('--subtask_report', type=str,
+							help='file to store report from subtask validation (in markdown)')
+		parser.add_argument('--program_report', type=str,
+							help='file to store report from program executions (in markdown)')
```

Comment:
What do you think about removing those arguments? (open for discussion)

My though process:
- `--verbose` is not clear what it means (it usually means that the CLI tool prints more data about what it's currently doing) and I doubt that the current name is intuitive with the current behavior (it expands the printed table). Because sometimes a solution isn't working as expected, the user behavior would probably be "running it without this flag, then seeing that something is wrong and rerunning the tool with this flag", which is a waste of time for the user when the solutions don't check quickly. Instead, I think that it's reasonable to print both a short table (like without this flag) and under it print a long table (like with this flag), so that the user can choose whether he wants to look into the short table or it's necessary to look into the results of the long table. Printing both tables at once can be done in a separate PR, it's not important to do it now.
- `--show_memory` should be enabled by default and there isn't really any reason why someone would want to disable it, besides when `oiejq` doesn't work with printing the memory. But in such a case, it might be better to detect that printing the memory doesn't work and somehow cope with that.
- `--validate_subtasks` and `--validate_programs` should always be enabled, the tool should first run all the solutions and then check whether a solution scored a different number of points than expected.
- `--subtask_report` and `--program_report` probably should be merged together and enabled by default, so that the user has the latest report in a saved file (just like `raport.html` in sinol packages -- we can try to replace this file with those reports). That would also solve the fact of missing/forgetting creating the reports (though, my proposal of `sinol_expected_scores` already mostly solves this issue, thus I'm leaving it open for discussion).

I'm basing my propositions on two rules:

It's good to polish the available commands and their names now, when we haven't yet written tests for them (so that we won't have to change them later). If we're going to need one of the flags, we can always add it later -- from the user experience, it's much easier to add functionality instead of changing it. Polishing the descriptions of command can be made later (which we will do -- clear command descriptions and a clear `--help` message is critical for pleasant usage of the tool).

The user should have reasonable default arguments that checks whether the user makes no mistakes as much as possible. If there is no reason to provide the user with a mechanism that disables one of the checks, we shouldn't do it. Thanks to that, the user will have to look into more detail into why a check failed, instead of ignoring it and running the tool without a check.

Reply:
``verbose``: I agree that verbose here doesn't mean what it usually means, we should delete id

Reply:
``show_memory``: I agree, but is there a possibility that oiejq doesn't measure memory?

Reply:
> show_memory: I agree, but is there a possibility that oiejq doesn't measure memory?

Yes, on Arch Linux often there are problems with it. We'll see whether that's indeed the case in step 1c (tests).

Reply:
> but I don't think we should replace raport.html

What ideas do you have then?

Reply:
Maybe we could change the current behavior so that it generates ``raport.html``?

Reply:
Let me know what you think about this approach: when the user runs the tool and it prints a table (that shows for each solution got how many points on each test), assuming that the user checks all of the solutions on all of the tests, the tool can automatically save the same table to `raport.html` (if we're going to make so that the tool shows both short and long tables during `sinol-make run`, we can also put both the tables into this file).

Reply:
Yes, I think that's a good idea

Reply:
By default when ``--validate_subtasks`` and ``--validate_programs`` are both not set, the program runs subtasks and programs. Should I leave it like that, or do we want to delete these flags?

Reply:
My plan was to remove the "subtasks" check and instead have the one I proposed in the docs. Then the flags won't be at all needed. I'm open for suggestions though.

Reply:
New subtask config: ac0accae2faedc5c318dcea52fec683f9721aba0
Suggest subtask configuration: 63ac565f28695a63e4c5679382a33d08d91064a6